internal/rootcoord/drop_collection_task.go

```diff
@@ -74,6 +74,7 @@ func (t *dropCollectionTask) Execute(ctx context.Context) error {
 	redoTask.AddAsyncStep(&dropIndexStep{
 		baseStep: baseStep{core: t.core},
 		collID:   collMeta.CollectionID,
+		partID:   -1,
```

Comment:
According to the gc logic of indexcoord, passing `-1` will lead indexcoord filter everything, so nothing will be collected. Just passing `nil` here.